rust/src/qfunction.rs

```diff
@@ -140,22 +191,85 @@ impl QFunction {
     pub fn create(
         ceed: &crate::Ceed,
         vlength: i32,
-        f: bind_ceed::CeedQFunctionUser,
+        f: QFunctionUserClosure,
         source: impl Into<String>,
     ) -> Self {
         let source_c = CString::new(source.into()).expect("CString::new failed");
         let mut ptr = std::ptr::null_mut();
+
+        let number_inputs = 0;
+        let number_outputs = 0;
+        let input_sizes = [0; crate::MAX_QFUNCTION_FIELDS];
+        let output_sizes = [0; crate::MAX_QFUNCTION_FIELDS];
+        let trampoline_data = QFunctionTrampolineData {
+            number_inputs,
+            number_outputs,
+            input_sizes,
+            output_sizes,
+        };
+
+        let qf_closure = |q: bind_ceed::CeedInt,
```

Comment:
@jedbrown @YohannDudouit 

This isn't quite right but should be close to what we need for (finally) finish this. Thoughts?